src/Pheanstalk.php

```diff
@@ -258,6 +258,27 @@ public function reserve(): Job
         return new Job($response['id'], $response['jobdata']);
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function reserveJob(JobIdInterface $job): Job
+    {
+        // New in 1.12, beanstalkd returns BadFormat instead of UnknownCommand
+        try {
+            $response = $this->dispatch(
+                new Command\ReserveJobCommand($job)
+            );
+        } catch (Exception\ServerBadFormatException $e) {
+            throw new Exception\ServerUnknownCommandException();
+        }
+
+        if ($response->getResponseName === ResponseInterface::RESPONSE_BAD_FORMAT) {
```

Comment:
This seems to be a typo, `$response->getResponseName` is a function not a property.

More importantly this should be caught by tests. Or even static analysis. (This is for my own todo list #223 )

Reply:
It's a typo. Test didn't catch it since the __get() on ArrayResponse swallows the warning here and coalesces it to null.